cli.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package celvet
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	api "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	apiinstall "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/install"
+	apiv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	structuralschema "k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	"k8s.io/apimachinery/pkg/runtime"
+	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+func Lint(args []string, outputWriter io.Writer) (int, error) {
+	if len(args) != 2 {
+		return 1, fmt.Errorf("usage: %s crd-file", args[0])
+	}
+
+	crdFile := args[1]
+	fileBytes, err := ioutil.ReadFile(crdFile)
+	if err != nil {
+		return 1, fmt.Errorf("error reading %s: %s", crdFile, err)
+	}
+
+	scheme := runtime.NewScheme()
+	apiinstall.Install(scheme)
+	codecs := runtimeserializer.NewCodecFactory(scheme)
+	decode := codecs.UniversalDeserializer().Decode
+	obj, _, err := decode(fileBytes, nil, nil)
+	if err != nil {
+		return 1, fmt.Errorf("error while decoding: %s", err)
+	}
+	switch obj.(type) {
+	case *apiv1.CustomResourceDefinition:
+	default:
+		return 1, fmt.Errorf("unexpected decoded object (expected CustomResourceDefinition), got %T", obj)
+	}
+
+	spec := obj.(*apiv1.CustomResourceDefinition).Spec
+	// TODO(DangerOnTheRanger): support multiple CRD versions
+	v1Schema := spec.Versions[0].Schema.OpenAPIV3Schema
```

Comment:
Just loop through them all here? What happens if two versions have errors on the same field? Is there a way to know which version the error was caused by?

Reply:
For some reason I was previously thinking each individual linter pass would need to be aware of the CRD version, but I don't think that's true anymore. Maybe just printing out the version as a header (`errors found in version 0:`) would be good, but perhaps there's a better UX option out there..

Reply:
Providing the path from the root of the CRD for errors, roughtly like `spec.versions[name=v1alpha1].schema.openAPIV3Schema.foo.baz` seems sufficiently clear (check the validation output for the exact syntax). It leaves no room for confusion about what exact field an error is for.  This is both consistent with how the validation output works and it's easily machine readable.

Reply:
Gotcha. I'll mark this as resolved once I add that.